src/lib/components/workspace/member-info-popup/MemberInfoPopup.svelte

```diff
@@ -3,35 +3,50 @@
   import MemberDropdown from "$lib/components/dropdown/MemberDropdown.svelte";
   import { fly, fade } from "svelte/transition";
   import MemberWorkspace from "../member-worspace/MemberWorkspace.svelte";
-  import { TeamRole } from "$lib/utils/enums/team.enum";
+  import { TeamRole, WorkspaceRole } from "$lib/utils/enums/team.enum";
   import type {
     TeamRepositoryMethods,
     TeamServiceMethods,
+    userDetails,
   } from "$lib/utils/interfaces";
 
-  export let title;
-  export let user;
-  export let workspaces;
-  export let onCancel;
-  export let userType;
-  export let handleMemberPromotePopUpCancel;
-  export let handleMemberDemotePopUpCancel;
-  export let handleMemberPopUpCancel;
-  export let handleMemberOwnershipPopUpCancel;
+  export let title: string;
+  export let user: userDetails;
+  export let teamRole: TeamRole = null;
+  export let workspaces: any[];
+  export let hasPermission: boolean = false;
+  export let onCancel: (flag: boolean) => void;
+  export let userType: TeamRole;
+  export let handleMemberPromotePopUpCancel = (flag: boolean) => {};
+  export let handleMemberDemotePopUpCancel = (flag: boolean) => {};
+  export let handleMemberPopUpCancel = (flag: boolean) => {};
+  export let handleMemberOwnershipPopUpCancel = (flag: boolean) => {};
   export let owner: boolean = false;
-  export let teamServiceMethods: TeamServiceMethods;
+  export let teamServiceMethods: TeamServiceMethods = {};
   export let userId: string;
-  export let handleMemberPopUpSuccess;
+  export let handleMemberPopUpSuccess = (flag: boolean) => {};
   export let getPermissionsData;
-
+  export let isWorkspaceMemberInfo = false;
+  export let handleDropDownWorkspaceLevel = (
+    currentRole: WorkspaceRole | "remove",
+  ) => {};
   const handleDropdown = (id) => {
     if (id === "remove") {
       handleMemberPopUpCancel(true);
-    } else if (user.role === TeamRole.ADMIN && id === TeamRole.MEMBER) {
+    } else if (
+      user.role === TeamRole.TEAM_ADMIN &&
+      id === TeamRole.TEAM_MEMBER
+    ) {
       handleMemberDemotePopUpCancel(true);
```

Comment:
use the same function then once the unified function is created with aditional param